expyfun/_input_controllers.py

```diff
@@ -157,18 +167,29 @@ def check_force_quit(self, keys=None):
         if len(keys):
             raise RuntimeError('Quit key pressed')
 
-    def _correct_presses(self, pressed, timestamp, relative_to):
+    def _correct_presses(self, pressed, timestamp, relative_to,
+                         releases=False):
         """Correct timing of presses and check for quit press"""
+        correctedEvents = []
         if len(pressed):
-            pressed = [(k, s + self.time_correction) for k, s in pressed]
+            pressed = [(k, s + self.time_correction, r) for k, s, r in pressed]
             self.log_presses(pressed)
-            keys = [k for k, _ in pressed]
+            keys = [k[0] for k in pressed]
             self.check_force_quit(keys)
-            if timestamp:
-                pressed = [(k, s - relative_to) for k, s in pressed]
-            else:
-                pressed = keys
-        return pressed
+            for event in pressed:
+                correctedEvent = [event[0]]
+                if timestamp:
+                    correctedEvent.append(event[1] - relative_to)
+                if releases:
+                    correctedEvent.append(event[2])
+                elif event[2] == 'release':
+                    continue  # removes releases.
+                if len(correctedEvent) > 1:
+                    correctedEvent = tuple(correctedEvent)
+                else:
+                    correctedEvent = correctedEvent[0]
+                correctedEvents.append(correctedEvent)
+        return correctedEvents
```

Comment:
e.g., this passage I think could be simplified by standardizing the `get_presses` return signature.